tensorboard/webapp/metrics/views/card_renderer/scalar_card_test.ts

```diff
@@ -3592,9 +3561,158 @@ describe('scalar card', () => {
+        const fixture = createComponent('card1');
+        fixture.detectChanges();
+
+        const testController = fixture.debugElement.query(
+          By.directive(CardFobControllerComponent)
+        ).componentInstance;
+
+        testController.onPrespectiveStepChanged.emit(20);
+        fixture.detectChanges();
+        testController.prospectiveFobClicked(new MouseEvent('mouseclick'));
+        fixture.detectChanges();
+
+        store.overrideSelector(selectors.getMetricsStepSelectorEnabled, true);
+        store.refreshState();
+        fixture.detectChanges();
+
+        expect(
+          fixture.componentInstance.stepSelectorTimeSelection$.getValue()
+        ).toEqual({
+          start: {step: 20},
+          end: null,
+        });
+      }));
+
+      it('removes step selection when step selection disabled', fakeAsync(() => {
+        const fixture = createComponent('card1');
+        fixture.detectChanges();
+
+        store.overrideSelector(selectors.getMetricsStepSelectorEnabled, true);
+        store.refreshState();
+        fixture.detectChanges();
+
+        expect(
+          fixture.componentInstance.stepSelectorTimeSelection$.getValue()
+        ).toEqual({
+          start: {step: 10}, // Sets to min.
+          end: null,
+        });
+
+        store.overrideSelector(selectors.getMetricsStepSelectorEnabled, false);
+        store.refreshState();
+        fixture.detectChanges();
+
+        expect(
+          fixture.componentInstance.stepSelectorTimeSelection$.getValue()
+        ).toEqual(null);
+      }));
+
+      it('adds end when range selection enabled', fakeAsync(() => {
+        const fixture = createComponent('card1');
+        fixture.detectChanges();
+
+        const testController = fixture.debugElement.query(
+          By.directive(CardFobControllerComponent)
+        ).componentInstance;
+
+        testController.onPrespectiveStepChanged.emit(20);
+        fixture.detectChanges();
+        testController.prospectiveFobClicked(new MouseEvent('mouseclick'));
+        fixture.detectChanges();
+
+        store.overrideSelector(selectors.getMetricsRangeSelectionEnabled, true);
+        store.refreshState();
+        fixture.detectChanges();
+
+        expect(
+          fixture.componentInstance.stepSelectorTimeSelection$.getValue()
+        ).toEqual({
+          start: {step: 20},
+          end: {step: 30}, // Sets to end.
+        });
+      }));
+
+      it('removes end when range selection disabled', fakeAsync(() => {
+        const fixture = createComponent('card1');
+        fixture.detectChanges();
+
+        const testController = fixture.debugElement.query(
+          By.directive(CardFobControllerComponent)
+        ).componentInstance;
+
+        testController.onPrespectiveStepChanged.emit(20);
+        fixture.detectChanges();
+        testController.prospectiveFobClicked(new MouseEvent('mouseclick'));
+        fixture.detectChanges();
```

Comment:
This pattern is used enough to warrant an addNewFobFromChart(step) utility function.

Reply:
Done.